Source/Scene/Batched3DModel3DTileContent.js

```diff
@@ -0,0 +1,455 @@
+            get: function() {
+                return this._url;
+            }
+        },
+
+        /**
+         * Part of the {@link Cesium3DTileContent} interface.
+         */
+        batchTable : {
+            get : function() {
+                return this._batchTable;
+            }
+        }
+    });
+
+    var sizeOfUint32 = Uint32Array.BYTES_PER_ELEMENT;
+
+    function getBatchIdAttributeName(gltf) {
+        var batchIdAttributeName = getAttributeOrUniformBySemantic(gltf, '_BATCHID');
+        if (!defined(batchIdAttributeName)) {
+            batchIdAttributeName = getAttributeOrUniformBySemantic(gltf, 'BATCHID');
+            if (defined(batchIdAttributeName)) {
+                Batched3DModel3DTileContent._deprecationWarning('b3dm-legacy-batchid', 'The glTF in this b3dm uses the semantic `BATCHID`. Application-specific semantics should be prefixed with an underscore: `_BATCHID`.');
+            }
+        }
+        return batchIdAttributeName;
+    }
+
+    function getVertexShaderCallback(content) {
+        return function(vs) {
+            var batchTable = content._batchTable;
+            var gltf = content._model.gltf;
+            var batchIdAttributeName = getBatchIdAttributeName(gltf);
+            var callback = batchTable.getVertexShaderCallback(true, batchIdAttributeName);
+            return defined(callback) ? callback(vs) : vs;
+        };
+    }
+
+    function getPickVertexShaderCallback(content) {
+        return function(vs) {
+            var batchTable = content._batchTable;
+            var gltf = content._model.gltf;
+            var batchIdAttributeName = getBatchIdAttributeName(gltf);
+            var callback = batchTable.getPickVertexShaderCallback(batchIdAttributeName);
+            return defined(callback) ? callback(vs) : vs;
+        };
+    }
+
+    function getFragmentShaderCallback(content) {
+        return function(fs) {
+            var batchTable = content._batchTable;
+            var gltf = content._model.gltf;
+            var diffuseUniformName = getAttributeOrUniformBySemantic(gltf, '_3DTILESDIFFUSE');
+            var callback = batchTable.getFragmentShaderCallback(true, diffuseUniformName);
+            return defined(callback) ? callback(fs) : fs;
+        };
+    }
+
+    function initialize(content, arrayBuffer, byteOffset) {
+        var tileset = content._tileset;
+        var tile = content._tile;
+        var basePath = getAbsoluteUri(getBaseUri(content._url, true));
+
+        var byteStart = defaultValue(byteOffset, 0);
+        byteOffset = byteStart;
+
+        var uint8Array = new Uint8Array(arrayBuffer);
+        var view = new DataView(arrayBuffer);
+        byteOffset += sizeOfUint32;  // Skip magic
+
+        //>>includeStart('debug', pragmas.debug);
+        var version = view.getUint32(byteOffset, true);
+        if (version !== 1) {
+            throw new DeveloperError('Only Batched 3D Model version 1 is supported.  Version ' + version + ' is not.');
```

Comment:
This should be a RuntimeError and not be pragma'd. Generic viewers don't have control over the type of content their users load and need to have errors like this bubble up the chain and be handled.